src/main/java/taxi/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,61 @@
+package taxi.dao;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import taxi.lib.Dao;
+import taxi.model.Car;
+import taxi.model.Driver;
+import taxi.storage.Storage;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        Storage.addCar(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        return Optional.ofNullable(Storage.getCars().stream()
+                .filter(car -> car.getId().equals(id))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Car with id " + id + " not find")));
```

Comment:
```suggestion
                .findFirst();
```
there is no point in returning optional if you throw an exception when the object is not found